front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -76,6 +79,12 @@ export default function VaultSideBarMenu({ owner }: VaultSideBarMenuProps) {
           );
         })}
       </Item.List>
+      <CreateVaultModal
```

Comment:
I'm not convince that this should live here. IMO it should be in the VaultLayout component and watches the router.

Reply:
LGTM